src/utilcode/longfilepathwrappers.cpp

```diff
@@ -1280,11 +1280,16 @@ BOOL PAL_GetPALDirectoryWrapper(SString& pbuffer)
 
 BOOL PAL_GetPALDirectoryWrapper(SString& pbuffer)
 {
-    BOOL retval;
+    BOOL retval = FALSE;
     COUNT_T size  = MAX_LONGPATH;     //Retry once the PAL Api is fixed to return the correct size 
     WCHAR* buffer = pbuffer.OpenUnicodeBuffer(size - 1);
 
-    retval = PAL_GetPALDirectoryW(pbuffer.OpenUnicodeBuffer(size - 1), size);
+    if(! PAL_GetPALDirectoryW(pbuffer.OpenUnicodeBuffer(size - 1), size))
+    {
+        pbuffer.CloseBuffer(0);
+        retval = PAL_GetPALDirectoryW(pbuffer.OpenUnicodeBuffer(size - 1) , size);
```

Comment:
What is the point of trying again? The size doesn't change, since it is passed by value to the PAL_GetPALDirectoryW.